### DISCUSSION TITLE
Copy of DNG to JPEG

### DISCUSSION THREAD

**Discussion by daverupp** [libvips/libvips#2090] [Q&A]
Hello @jcupitt :-)

I wrote once a question about ProRAW format from the new iOS devices. I found out, that the format will be in the end a *.dng adobe raw format.

So, I am generating thumbnails out of it via the vipsthumbnail. that is working perfectly so far :)

What I tried out now was, to copy the dng to jpeg the same way I am doing it with *.heic to *.jpeg.
vips copy ship.heic ship_output.jpg
(see attached heic image)

That is working perfect. Also the exif infos are there etc...

When I try now the same thing with the *.dng file, I only get a very compressed jpeg without any exif info etc...

vips copy thing.dng thing_output.jpg

Do I have to copy the dng differently compared to the heic file to get a useable jpeg version?

thx a lot for your help.
Dave 

[images.zip](https://github.com/libvips/libvips/files/5701977/images.zip)

**Comment by daverupp** [comment]
anyone any idea about the upper question?

**Comment by jcupitt** [comment]
Hello @daverupp, sorry, this slipped off my to-do list.

DNG files are actually small JPEGs, with the actual pixel data of the raw camera sensor hidden in some extra metadata. You can open them with anything that can read JPEG, but you'll just see the thumbnail.

You might be able to read the files with imagemagick, if you configure it carefully. For example:

```
vips magickload thing.dng x.jpg
```

But you'll probably need to poke about inside imagemagick to get it working.

A better solution would be a libraw loader for libvips, as we've discussed elsewhere I think.

**Comment by daverupp** [comment]
Hello @jcupitt 
Do you know, what the problem here could be when I try to create a thumbnail out of the attached *.dng file? 
I see, that the original file format was an *.ORF format and then exported somehow via lightroom... (I assume...)
(It is done via an olympus camera)

I am getting a thumbnail, but it is super low-res... do I need to set another parameter ?
--size=4096x1080 tmp.dng -o tmp_output.jpg[optimize_coding,strip]

[tmp.dng.zip](https://github.com/libvips/libvips/files/5815362/tmp.dng.zip)

with other dng formats (like from an iphone pro) it is working very good, and the thumbnail is in good quality...
thx for looking into it.
cheers dave

**Comment by jcupitt** [comment]
Yes, you can't thumbnail DNGs, it just won't work. They are not standfard images, you have to use a specialized converter.

**Comment by daverupp** [comment]
But funny thing is, for the iphone dng images in the first zip from the original post, it is working the same way and it looks good with the quality.

**Comment by jcupitt** [comment]
DNGs are just JPG images with the raw camera data hidden in a private field, so the thumbnail you get is as good as the DNG thumbnail.

It sounds like your iphone is using a fairly large JPG as the main image, but the olympus is not.

**Comment by daverupp** [comment]
Ahh ok. Thx for the quick answer. Maybe I have to use dcraw then for normal cameras and for the iphone ones just as it is. 
all the best.

**Comment by Panda92524** [comment]
Hello
If so how to convert JPEG to RAW(DNG) files?
Please help me. This is urgent for me
thanks

**Reply by jcupitt** [reply]
I would use libraw, or its command-line tool `simple-dcraw`.

**Comment by kleisauke** [comment]
A dedicated LibRaw loader is now available on the master branch (which will be part of the upcoming 8.18 release).
